src/app/ngrx/reducers/podcast-downloads.reducer.ts

```diff
@@ -0,0 +1,77 @@
+import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
+import { ActionTypes, AllActions } from '../actions'
+import { PodcastDownloads } from './models/podcast-downloads.model';
+import { UpdateStr } from '@ngrx/entity/src/models';
+import * as includes from 'array-includes';
+
+export interface PodcastDownloadsState extends EntityState<PodcastDownloads>{};
+
+export const adapter: EntityAdapter<PodcastDownloads> = createEntityAdapter<PodcastDownloads>(); 
+
+export const initialState: EntityState<PodcastDownloads> = adapter.getInitialState();
+
+export function PodcastDownloadsReducer(state: PodcastDownloadsState = initialState, action: AllActions) {
+  switch (action.type) {
+    case ActionTypes.CASTLE_PODCAST_DOWNLOADS_LOAD: {
+      const { id } = action.payload;
+      const currentIds = <string[]>selectPodcastDownloadsIds(state);
+      const charted = includes(currentIds, id);
+      const podcastUpdate: UpdateStr<PodcastDownloads> = {
+        id,
+        changes: {
+          error: null,
+          loading: true,
+          loaded: false,
+          ...(charted && { charted }) // Conditionally add charted as object member
+        }
+      }
+      return adapter.upsertOne(podcastUpdate, state);
+    }
+    case ActionTypes.CASTLE_PODCAST_DOWNLOADS_SUCCESS: {
+      const { id, metricsPropertyName, metrics } = action.payload;
+      const currentIds = <string[]>selectPodcastDownloadsIds(state);
+      const charted = includes(currentIds, id);
```

Comment:
One thing I see that's changed is that if you toggle off the podcast from the stacked chart, then change the date range or page, the podcast becomes charted again. In production, the podcast chart is default toggled on when first loaded but stays toggled off if you change the parameters. Here it's charting it by default always util toggled off.

Reply:
D'oh! Accidentally flipped that logic. Pushed a fix.